lib/paperclip/file_command_content_type_detector.rb

```diff
@@ -14,9 +14,9 @@ def detect
 
     def type_from_file_command
       type = begin
-        # On BSDs, `file` doesn't give a result code of 1 if the file doesn't exist.
-        Paperclip.run("file", "-b --mime :file", :file => @filename)
-      rescue Cocaine::CommandLineError => e
+        MimeMagic.by_magic(File.open(@filename)).try(:type) ||
+          Paperclip.run("file", "-b --mime :file", :file => @filename)
```

Comment:
Since this method is `type_by_file_command` it would make sense to extract this out into another method.